src/dvclive/live.py

```diff
@@ -299,6 +301,19 @@ def log_param(
         """Saves the given parameter value to yaml"""
         self.log_params({name: val})
 
+    def log_artifact(
+        self,
+        path: str,
+    ):
+        """Saves the given parameter value to yaml"""
+        if self._dvc_repo is not None and not self._inside_dvc_exp:
```

Comment:
We are going to encounter lock issues here if we are inside `dvc repro` execution. 
Probably need to add a new property `self._inside_dvc_repro` and a way to infer this.